flow360/component/flow360_params/validations.py

```diff
@@ -37,3 +49,176 @@ def _check_duplicate_boundary_name(values):
                 )
             boundary_names.add(boundary_name_curr)
     return values
+
+
+def _check_consistency_wall_function_and_surface_output(values):
+    boundary_types = []
+    boundaries = values.get("boundaries")
+    if boundaries is not None:
+        boundary_types = boundaries.get_subtypes()
+
+    surface_output_fields_root = []
+    surface_output = values.get("surfaceOutput")
+    if surface_output is not None:
+        surface_output_fields_root = surface_output.output_fields
+    if "WallFunctionMetric" in surface_output_fields_root and WallFunction not in boundary_types:
+        raise ValueError(
+            "'WallFunctionMetric' in 'surfaceOutput' is only valid for 'WallFunction' boundary types."
+        )
+    return values
+
+
+def _check_consistency_ddes_volume_output(values):
+    turbulence_model_solver = values.get("turbulence_model_solver")
+    model_type = None
+    run_ddes = False
+    if turbulence_model_solver is not None:
+        model_type = turbulence_model_solver.model_type
+        run_ddes = turbulence_model_solver.DDES
+
+    volume_output = values.get("volumeOutput")
+    if volume_output is not None and volume_output.output_fields is not None:
+        output_fields = volume_output.output_fields
+        if "SpalartAllmaras_DDES" in output_fields and not (
+            model_type == "SpalartAllmaras" and run_ddes
+        ):
+            raise ValueError(
+                "SpalartAllmaras_DDES output can only be specified with \
+                SpalartAllmaras turbulence model and DDES turned on"
+            )
+        if "kOmegaSST_DDES" in output_fields and not (model_type == "kOmegaSST" and run_ddes):
+            raise ValueError(
+                "kOmegaSST_DDES output can only be specified with kOmegaSST turbulence model and DDES turned on"
+            )
+    return values
+
+
+def _validate_cht_has_heat_transfer_zone(values):
+    heat_equation_solver = values.get("heat_equation_solver")
```

Comment:
we need to make sure this validator runs after alias inference, i.e., after heatEquationSolver is converted to heat_equation_solver. Not sure at  which stage alias are applied.

Reply:
I will add some unit tests to confirm